src/main/java/opendcs/dao/AlgorithmDAO.java

```diff
@@ -54,432 +23,493 @@
-			{
-				propertiesSqlDao.deleteProperties("CP_ALGO_PROPERTY", "ALGORITHM_ID", id);
-			}
-			
-			q = "delete from CP_ALGO_SCRIPT where ALGORITHM_ID = " + id;
-			doModify(q);
-
-			q = "delete from CP_ALGORITHM where ALGORITHM_ID = " + id;
-			doModify(q);
-		}
-		catch(SQLException ex)
-		{
-			String msg = "Error deleting algorithm with ID=" + id + ": " + ex;
-			warning(msg);
-			throw new DbIoException(msg);
-		}
-	}
-	
-	@Override
-	public void close()
-	{
-		super.close();
-	}
-
-	@Override
-	public ArrayList<String> listAlgorithmNames() throws DbIoException
-	{
-		String q = "select ALGORITHM_NAME from CP_ALGORITHM";
-		try
-		{
-			ResultSet rs = doQuery(q);
-			ArrayList<String> ret = new ArrayList<String>();
-			while(rs.next())
-				ret.add(rs.getString(1));
-			return ret;
-		}
-		catch(SQLException ex)
-		{
-			String msg = "Error listing algorithms: " + ex;
-			warning(msg);
-			throw new DbIoException(msg);
-		}
-	}
-
-	@Override
-	public ArrayList<AlgorithmInList> listAlgorithmsForGui()
-		throws DbIoException
-	{
-		String q = "select algorithm_id, algorithm_name, "
-			+ "exec_class, cmmnt "
-			+ "from cp_algorithm";
-		try
-		{
-			ResultSet rs = doQuery(q);
-			ArrayList<AlgorithmInList> ret = new ArrayList<AlgorithmInList>();
-			while(rs != null && rs.next())
-				ret.add(new AlgorithmInList(DbKey.createDbKey(rs, 1), rs.getString(2),
-					rs.getString(3), 0, TextUtil.getFirstLine(rs.getString(4))));
-			
-			q = "select a.algorithm_id, count(1) as CompsUsingAlgo "
-				+ "from cp_algorithm a, cp_computation b "
-				+ "where a.algorithm_id = b.algorithm_id "
-				+ "group by a.algorithm_id";
-			rs = doQuery(q);
-			while(rs != null && rs.next())
-			{
-				DbKey algoId = DbKey.createDbKey(rs, 1);
-				int numCompsUsing = rs.getInt(2);
-				for(AlgorithmInList ail : ret)
-				{
-					if (ail.getAlgorithmId().equals(algoId))
-					{
-						ail.setNumCompsUsing(numCompsUsing);
-						break;
-					}
-				}
-			}
-		
-			return ret;
-		}
-		catch(SQLException ex)
-		{
-			String msg = "Error listing algorithms for GUI: " + ex;
-			warning(msg);
-			throw new DbIoException(msg);
-		}
-	}
+    public static final org.slf4j.Logger log = LoggerFactory.getLogger(AlgorithmDAO.class);
+
+    // While it *is* possible for this to change during runtime. The cost of
+    // a database query is not trivial so we will have a default here, attempt
+    // to determine it once per application instance, and except the result for
+    // the length of that run.
+    public static Integer COMMENT_LENGTH = null;
+
+    public AlgorithmDAO(DatabaseConnectionOwner tsdb)
+    {
+        super(tsdb, "AlgorithmDao");
+        if (COMMENT_LENGTH == null)
+        {
+            COMMENT_LENGTH = 1000;
+            try(DaoBase dao = new DaoBase(tsdb, module))
+            {
+                dao.doQuery("select cmmnt from cp_algorithm", rs ->
+                {
+                    ResultSetMetaData rsmd = rs.getMetaData();
```

Comment:
Will this code re-read the meta-data for each row?    

consider   dao.getFirstResult() instead.

Reply:
oh that's a good point.